tests/test_user_document.py

```diff
@@ -62,35 +58,36 @@ def test_post_patient_report(self):
         create_service_token(client=client, user=get_user(TEST_USER_ID))
         self.login()
 
-        test_contents = "This is a test."
+        test_contents = 'This is a test.'
         with NamedTemporaryFile(
             prefix='udoc_test_',
             suffix='.pdf',
             delete=True,
         ) as temp_pdf:
-            temp_pdf.write(test_contents)
+            temp_pdf.write(test_contents.encode('utf-8'))
```

Comment:
I don't think you should need to do this since you're importing `unicode_literals`...

Reply:
I couldn't find an easy way to fix this without specifying the encoding somewhere, but I realized that creating the `NamedTemporaryFile` is entirely unnecessary.

We're writing a string to `temp_pdf`, then just reading it into a `BytesIO`; ie `self.client.post` doesn't even reference `temp_pdf` (except for a filename). Can you remove the `NamedTemporaryFile` block and just create the file with `StringIO`? eg:

```
test_contents = 'This is a test.'
response = self.client.post(
    '/api/user/{}/patient_report'.format(TEST_USER_ID),
    content_type='multipart/form-data', 
    data=dict({'file': (StringIO(test_contents), 'udoc_test.pdf')}),
)

```

Reply:
Good catch on removing the `NamedTemporaryFile`, but I'm still getting the same error - this time it's on the `StringIO(test_contents)` line.

`nose.proxy.TypeError: 'str' does not support the buffer interface`

Aren't we writing a string with `StringIO` this way, too (and thus have to encode it)?

Reply:
Looks like you (should NOT need it in python3)[https://stackoverflow.com/a/9368909] but will in py2.

Reply:
Please try [Martijn Pieters solution](https://stackoverflow.com/a/34872005) via imports.

Reply:
Good call, I'll take test_user_document into its own branch and try that solution.

Reply:
So that seems to be working for the `StringIO` bit, but now I'm getting an error on the line with `'udoc_test.pdf'` in this block:

        `response = self.client.post(
            '/api/user/{}/patient_report'.format(TEST_USER_ID),
            content_type='multipart/form-data',
            data=dict({'file':
                           (StringIO(test_contents),
                            'udoc_test.pdf')})
        )`

error: `nose.proxy.TypeError: 'str' does not support the buffer interface`
But if I encode `'udoc_test.pdf'` as a bytestring, then I get the error: `nose.proxy.TypeError: can't use a string pattern on a bytes-like object`

Any idea what is going on?